polars/polars-core/src/chunked_array/ops/explode.rs

```diff
@@ -49,60 +48,54 @@ impl ChunkExplode for Utf8Chunked {
         // of the list. And we also return a slice of the offsets. This slice can be used to find the old
         // list layout or indexes to expand the DataFrame in the same manner as the 'explode' operation
         let ca = self.rechunk();
-        let stringarr: &Utf8Array<i64> = ca
+        let array: &Utf8Array<i64> = ca
             .downcast_iter()
             .next()
             .ok_or_else(|| PolarsError::NoData("cannot explode empty str".into()))?;
-        let list_data = stringarr.data();
-        let str_values_buf = stringarr.value_data();
-
-        // We get the offsets of the strings in the original array
-        let offset_ptr = list_data.buffers()[0].as_ptr() as *const i64;
-        // offsets in the list array. These indicate where a new list starts
-        let offsets = unsafe { std::slice::from_raw_parts(offset_ptr, self.len()) };
+        let values = array.values();
+        let offsets = array.offsets();
 
         // Because the strings are u8 stored but really are utf8 data we need to traverse the utf8 to
         // get the chars indexes
-        let str_data = unsafe { std::str::from_utf8_unchecked(str_values_buf.as_slice()) };
+        // Utf8Array guarantees that this holds.
+        let str_data = unsafe { std::str::from_utf8_unchecked(values) };
+
         // iterator over index and chars, we take only the index
-        // todo! directly create a buffer from an aligned vec or a mutable buffer
-        let mut new_offsets = str_data.char_indices().map(|t| t.0 as i64).collect_vec();
-        // somehow I don't get the last value if we don't add this one.
```

Comment:
The somehow here is because the last offset in arrow must be the length of the values, while char_indices only gives the start position of the char.